kinesisfirehose-logs-extension-demo/README.md

```diff
@@ -0,0 +1,225 @@
+Built Template   : .aws-sam/build/template.yaml
+
+Commands you can use next
+=========================
+[*] Invoke Function: sam local invoke
+[*] Deploy: sam deploy --guided
+```
+
+### Deployment
+
+Run the following command to deploy the sample lambda function with the extension
+
+```bash
+sam deploy --guided
+```
+
+The following parameters can be customized part of the deployment
+
+| Parameter  | Description | Default |
+| ------------- | ------------- | ----|
+| FirehoseStreamName  | Firehose stream name |  lambda-logs-direct-s3-no-cloudwatch |
+| FirehoseS3Prefix  | The S3 Key prefix for Kinesis Firehose |  lambda-logs-direct-s3-no-cloudwatch |
+| FirehoseCompressionFormat  | Compression format used by Kinesis Firehose, allowed value - `UNCOMPRESSED, GZIP, Snappy` |  GZIP |
+| FirehoseBufferingInterval  | How long Firehose will wait before writing a new batch into S3 |  60 |
+| FirehoseBufferingSize  | Maximum batch size in MB |  10 |
+
+> Note: We can either customize the parameters, or leave it as default to proceed with the deployment
+
+**Output**
+
+```bash
+CloudFormation outputs from deployed stack
+-------------------------------------------------------------------------------------------------------------------
+Outputs
+-------------------------------------------------------------------------------------------------------------------
+Key                 KinesisFireHoseLogsApiExtensionLayer
+Description         Kinesis Log emiter Lambda Extension Layer Version ARN
+Value               arn:aws:lambda:us-east-1:xxx:layer:kinesisfirehose-logs-extension-demo:5
+
+Key                 BucketName
+Description         The bucket where data will be stored
+Value               sam-app-deliverybucket-xxxx
+
+Key                 KinesisFireHoseIamRole
+Description         Kinesis firehose IAM role
+Value               arn:aws:firehose:us-east-1:xxx:deliverystream/lambda-logs-direct-s3-no-cloudwatch
+
+Key                 HelloWorldFunction
+Description         First Lambda Function ARN
+Value               arn:aws:lambda:us-east-1:xxx:function:kinesisfirehose-logs-extension-demo-function
+-------------------------------------------------------------------------------------------------------------------
+```
+
+## Testing
+
+You can invoke the Lambda function using the following CLI command
+
+```bash
+aws lambda invoke \
+    --function-name "<<function-name>>" \
+    --payload '{"payload": "hello"}' /tmp/invoke-result \
+    --cli-binary-format raw-in-base64-out \
+    --log-type Tail
+```
+
+>Note: Make sure to replace `function-name` with the actual lambda function name
+
+The function should return ```"StatusCode": 200```, with the below output
+
+```bash
+{
+    "StatusCode": 200,
+    "LogResult": "<<Encoded>>",
+    "ExecutedVersion": "$LATEST"
+}
+```
+
+In a few minutes after the successfully invocation of the lambda function, we should start seeing the log messages from the example extension written to an S3 bucket.
```

Comment:
I think its worth expanding this slightly to say that the log messages are sent to Kinesis Data Firehose which sends the messages to S3.

Reply:
Updated